app/app_stack.py

```diff
@@ -170,55 +169,49 @@ def __init__(self, scope: Construct, id: str, **kwargs) -> None:
             retry_on_service_exceptions=True,
             payload=sfn.TaskInput.from_object(
                 {
-                    "date": sfn.JsonPath.string_at("$.date"),
-                    "hour": sfn.JsonPath.string_at("$.hour"),
+                    "zarr_store": sfn.JsonPath.string_at("$.zarr_store"),
                     "reply": sfn.JsonPath.object_at("$.CheckResult.Payload"),
                 }
             ),
         )
 
-        download_era5_task.add_retry(
+        send_request_task.add_retry(
             errors=["CDSAPITooManyRequests"],
             max_attempts=3,
             backoff_rate=2,
             interval=Duration.minutes(15),
         )
-        download_era5_task.add_retry(
+        send_request_task.add_retry(
             errors=["CDSAPINotAvailableYet"],
             max_attempts=3,
             backoff_rate=1,
             interval=Duration.days(1),
         )
-        download_era5_task.add_retry(
+        send_request_task.add_retry(
             errors=["States.ALL"],
             max_attempts=3,
             backoff_rate=2,
             interval=Duration.minutes(30),
         )
 
         # Logic flow & State
-        check_file_fail_state = sfn.Fail(
-            self,
-            "checkFileFail",
-            comment="Something went wrong checking if file exists!",
-        )
         check_file_success_state = sfn.Succeed(
-            self, "fileSuccess", comment="File exist"
+            self, "fileSuccess", comment="File already exist"
         )
-        checkfile_exists_state = sfn.Choice(
+        file_ok: sfn.Choice = sfn.Choice(
             self,
             "checkFileExist",
         )
-        check_file_task.next(checkfile_exists_state)
-        checkfile_exists_state.when(
-            sfn.Condition.number_equals("$.CheckFile.Payload.StatusCode", 200),
+        check_file_task.next(file_ok)
+        file_ok.when(
+            Condition.number_equals("$.CheckFile.Payload.status_code", 200),
             check_file_success_state,
         )
-        checkfile_exists_state.when(
-            sfn.Condition.number_equals("$.CheckFile.Payload.StatusCode", 404),
+        file_ok.when(
+            sfn.Condition.number_equals("$.CheckFile.Payload.status_code", 404),
             send_request_task,
-        )
-        checkfile_exists_state.otherwise(check_file_fail_state)
+        )  # OR CONTAINS WRONG DATA?!?!?!?!?1
```

Comment:
This comment is uninformative. Either, if the problem is fixed, remove comment. If the uncertainty remains, fix the problem!

Reply:
Dramatic and unnecessary, will be removed :)